### ISSUE TITLE
Tree View nodes do not indicate their positions in set

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3572]
## Describe the bug

Tree nodes do not indicate their levels, positions in the data set and data set sizes.

This functionality will be necessary when the complete set of available nodes is not present in the DOM due to dynamic loading.  Therefore, each node should have aria-level, aria-setsize, and aria-posinset specified so the screen reader user hears these values as they interact with the tree nodes.

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Read the tree nodes of any Tree View example using a screen reader
- Observe that the screen reader does not announce any information regarding the level of the treenode, the overall data set size and the treenode's position in the data set.

## Expected behavior

Use aria-level, aria-setsize, and aria-posinset as seen at https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by hippee-lee** [comment]
For `aria-setsize` it seems like Clarity will be able to provide the ability to set this but it will be up to the _consumer/api_ to provide a tree with the correct information in some cases. ClrTreeView does support use cases where the data displayed is recursive or has an unknown depth - do we have any guidance on this? 

For `aria-level` Clarity will need to add a service to generate and track levels properly as the nodes are generated.

**Comment by sheribyrne** [comment]
Definitely a defect, but not as high priority as the others - basically the tree view node equivalent of infinite scrolling where you don't know where the "end" is

**Comment by gnomeontherun** [comment]
This is probably resolved by #3569

**Comment by jeeyun** [comment]
Took another look and this issue is not resolved by #3569.

**Comment by gnomeontherun** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
